### DISCUSSION TITLE
Tips for HA configuration?

### DISCUSSION THREAD

**Discussion by timchkor** [appwrite/appwrite#1306] [Q&A] [answered]
Hello,

I am new on Appwrite, finding tons of great features saving my teams' efforts setting up B/E.

Considering scalable deployment of critical enterprise applications, is there any options for High Availability, and multi cluster set-up?

Thanks!

**Comment by eldadfux** [comment, marked as answer]
Hey @timchkor, you can learn more about how to use Appwrite in production here: https://appwrite.io/docs/production


From the docs:

> Appwrite was built with scalability in mind. Appwrite can scale both horizontally and vertically.
> 
> Appwrite uses a few containers to run, where each container has its job. Most of the Appwrite containers are stateless, and in order to scale them, all you need is to replicate them and setup a load balancer to distribute their load.
> 
> If you decide to set up a load balancer for a specific container, make sure that the containers that are trying to communicate with it are accessing it through a load balancer and not directly. All connections between Appwrite different containers are set using Docker environment variables.
> 
> There are three Appwrite containers that do keep their state. MariaDB, Redis, and InfluxDB containers are used for storing data, cache and pub/sub messaging, and usage stats (in this order). To scale them out, all you need to do is set up a standard cluster (same as you would with any other app using these technologies) according to your needs and performance.
